### PR TITLE
v2 endpoint to patch machine deployment

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: v2 endpoint to patch machine deployment

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
v2 endpoint to patch machine deployment: PATCH /api/v2/projects/{project_id}/clusters/{cluster_id}/machinedeployments/{machinedeployment_id}
```

### CODE REVIEW

File: pkg/handler/common/machine.go [hunk 1]
```diff
@@ -419,6 +424,101 @@ func ListMachineDeploymentMetrics(ctx context.Context, userInfoGetter provider.U
...
+	if err != nil {
+		return nil, fmt.Errorf("cannot patch node deployment: %v", err)
+	}
+
+	var patchedNodeDeployment *apiv1.NodeDeployment
+	if err := json.Unmarshal(patchedNodeDeploymentJSON, &patchedNodeDeployment); err != nil {
+		return nil, fmt.Errorf("cannot decode patched cluster: %v", err)
+	}
+
+	//TODO: We need to make the kubelet version configurable but restrict it to versions supported by the control plane
```
Comment (lsviben) [inline pkg/handler/common/machine.go]: is this TODO still relevant? From what I see we are checking if the version is compatible below

**Comment by zreigz** [comment]
@lsviben I had to rebase, PTAL

**Pull request merged**